src/entity/Entity.php

```diff
@@ -785,17 +786,34 @@ protected function broadcastMovement(bool $teleport = false) : void{
 				$this->spawnTo($player);
 			}
 		}else{
-			$this->server->broadcastPackets($this->hasSpawned, [MoveActorAbsolutePacket::create(
-				$this->id,
-				$this->getOffsetPosition($this->location),
-				$this->location->pitch,
-				$this->location->yaw,
-				$this->location->yaw,
-				(
-					//TODO: if the above hack for #4394 gets removed, we should be setting FLAG_TELEPORT here
-					($this->onGround ? MoveActorAbsolutePacket::FLAG_GROUND : 0)
-				)
-			)]);
+			$deltaLoc = $offsetPosition->subtractVector($this->getOffsetPosition($this->lastLocation));
+			$pk = new MoveActorDeltaPacket();
+			$pk->actorRuntimeId = $this->id;
+			//TODO: if the above hack for #4394 gets removed, we should be setting FLAG_TELEPORT here
+			$pk->flags = $this->onGround ? MoveActorDeltaPacket::FLAG_GROUND : 0;
+			if($deltaLoc->x != 0.0){
+				$pk->xPos = $offsetPosition->x;
+				$pk->flags |= MoveActorDeltaPacket::FLAG_HAS_X;
+			}
+			if($deltaLoc->y != 0.0){
+				$pk->yPos = $offsetPosition->y;
+				$pk->flags |= MoveActorDeltaPacket::FLAG_HAS_Y;
+			}
+			if($deltaLoc->z !== 0.0){
+				$pk->zPos = $offsetPosition->z;
+				$pk->flags |= MoveActorDeltaPacket::FLAG_HAS_Z;
+			}
+			if(($this->location->pitch - $this->lastLocation->pitch) !== 0.0){
```

Comment:
This subtraction is pointless. It's not telling you anything that an equality operator wouldn't.